### DISCUSSION TITLE
Does this application have a version for macOS

### DISCUSSION THREAD

**Discussion by lbeenguyen** [VLprojects/mediasoup-client-swift#32] [Q&A]
I’m having difficulty developing with WebRTC MediasoupClient on macOS. Is there a version available for macOS? Thanks!

**Comment by fedulvtubudul** [comment]
No, we haven't tried using [mediasoup-client-swift](https://github.com/VLprojects/mediasoup-client-swift) for MacOS, but in theory there should be no problems. Start from patching build.sh to include desktop platform and architectures instead of iOS.


If you've already started, please give more details what blocks you from going further.

**Reply by lbeenguyen** [reply]
I have modified the build file: https://github.com/lbeenguyen/mediasoup-client-swift/blob/master/build.sh.
After building, I have the mediasoupclient.xcframework, sdptransform.xcframework, and WebRTC.xcframework with content inside being macos-arm64_x86_64/....
I also modified the CMakeLists.txt: https://github.com/lbeenguyen/mediasoup-client-swift/blob/master/Mediasoup/dependencies/CMakeLists.txt.
And I updated the script for macOS to the following:
`set -e

FRAMEWORK_NAME="Mediasoup"
BUILD_DIR="${SRCROOT}/build/Mediasoup"
OUTPUT_DIR="${SRCROOT}/bin"

# Remove build directory if it exists

# Build the framework for both arm64 and x86_64 architectures for macOS
xcodebuild -project "${PROJECT_NAME}.xcodeproj" \
    -scheme "${FRAMEWORK_NAME}" \
    -configuration Release \
    -arch arm64 -arch x86_64 only_active_arch=no \
    -sdk "macosx" \
    -derivedDataPath "${BUILD_DIR}"

# Prepare output directory
mkdir -p ${OUTPUT_DIR}
if [ -d "${OUTPUT_DIR}/${FRAMEWORK_NAME}.xcframework" ]; then
    rm -rf "${OUTPUT_DIR}/${FRAMEWORK_NAME}.xcframework"
fi

# Create XCFramework for macOS
xcodebuild -create-xcframework \
    -framework ${BUILD_DIR}/Build/Products/Release/${FRAMEWORK_NAME}.framework \
    -output $OUTPUT_DIR/${FRAMEWORK_NAME}.xcframework
`

**Reply by lbeenguyen** [reply]
However, I am still encountering some errors while trying to build the Mediasoup Framework. Please help me.

**Reply by lbeenguyen** [reply]
![image](https://github.com/user-attachments/assets/b3e1053e-8b2a-4888-ba2f-1b35c62109e7)

**Reply by lbeenguyen** [reply]
I have added the CoreAudioTypes.framework to Frameworks and Libraries, but it still reports an error saying it cannot be found.

![image](https://github.com/user-attachments/assets/26dba507-52bc-4916-b441-10930c57707a)
